### PR TITLE
Fix the concurrency issue in the defaultExecute.go

### PR DESCRIPTION
**Pull request by runjiezhang**
## Description
The current code inside the `execute/defaultExecute.go` has concurrency issues under some scenarios. When we got errors from the Ansible command, the code would return from [this line](https://github.com/apenella/go-ansible/blob/master/execute/defaultExecute.go#L118). Meanwhile, `cmd.Wait()` would close the pipe. However, the reads from the pipe insides the go routine may not have completed yet. Therefore, it would cause the code to panic: `panic: send on closed channel
` in [line 86](https://github.com/apenella/go-ansible/blob/master/execute/defaultExecute.go#L86).

I am currently using this package in one of our project inside the go plugin. The panic will bring down the plugin which is not what we want.

Please take a look at the changes and let me know if you have any questions. Thank you. @apenella 

## Example
To reproduce the error, you can change the [`Connection`](https://github.com/apenella/go-ansible/blob/master/examples/simple-ansibleplaybook/simple-ansibleplaybook.go#L10) in the example to `ssh` and run the code. Then you will see the code panic out like this:
```
panic: send on closed channel

goroutine 18 [running]:
github.com/apenella/go-ansible/execute.(*DefaultExecute).Execute.func1(0xc0000a6920, 0x1159af2, 0x12, 0x1186480, 0xc0000ae558, 0xc00008c120, 0xc00008c0c0)
        /Users/runzhang/playground/go-ansible/execute/defaultExecute.go:86 +0xd7
created by github.com/apenella/go-ansible/execute.(*DefaultExecute).Execute
        /Users/runzhang/playground/go-ansible/execute/defaultExecute.go:76 +0x331
```

### CODE REVIEW

**Comment by apenella** [comment]
Hi @runjiezhang 
I do appreciate your collaboration. I was aware that there is a concurrency issue but I hadn't found a reliable scenario to reproduce the issue since I received your message.
What I see is that, to reproduce the issue, you need a failure on your playbook, as you pointed me, and is also needed to reuse ' AnsiblePlaybookCmd`to execute several ansible runs.

Thank you very much!

**Comment by runjiezhang** [comment]
Hi @apenella ,
Thank you for you reply. Yes, you are correct. So can we get the concurrency issue fixed? Because when playbook fails, we just want to handle the error message instead of a panic caused by concurrency issue, so that our service won't crash just because of the playbook failure. 

I've tested the changes and it worked fine to me. It may not be the optimal fix but please let me know if you have any better idea on how to fix the concurrency issue.

Thank you!

**Review by apenella — APPROVED** [review]

File: execute/defaultExecute.go [hunk 1]
```diff
@@ -73,6 +73,12 @@ func (e *DefaultExecute) Execute(command string, args []string, prefix string) e
 		return errors.New("(DefaultExecute::Execute) -> " + err.Error())
 	}
 
+	timeInit := time.Now()
+	err = cmd.Start()
+	if err != nil {
+		return errors.New("(DefaultExecute::Execute) -> " + err.Error())
+	}
+
```
[the remark this answers is not in the document]
Comment (runjiezhang) [reply execute/defaultExecute.go:81]: @apenella Here is the reason behind. Let's say we hit the [line 92](https://github.com/apenella/go-ansible/blob/master/execute/defaultExecute.go#L92) in the original code by some reasons. The function will return and the channels will be close by `defer`. However, the cmd inside the goroutine will keep trying to write to the `execDoneChan` or `execErrChan` which are closed already. Panic will be raised in this scenario. 

Meanwhile, according to the example here (https://github.com/go-cmd/cmd#go-cmdcmd), I think it's okay to start the command before the goroutine.

[the remark this answers is not in the document]
Comment (apenella) [reply execute/defaultExecute.go:81]: Hi @runjiezhang
It make sense what you say. Beside, cmd.Start() does not wait for command to be completed, then the output is not blocked.
Thank you very much for your response.

**Pull request merged**